src/Persistence/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -16,17 +16,15 @@ public static void AddPowerAppsPersistence(this IServiceCollection services, boo
         services.AddSingleton<IControlFactory, ControlFactory>();
 
         if (useDefaultTemplates)
-            services.AddSingleton<IControlTemplateStore, ControlTemplateStore>(LoadControlTemplateStore);
+            services.AddSingleton<IControlTemplateStore, ControlTemplateStore>(WithDefaultTemplates);
         else
-            services.AddSingleton<IControlTemplateStore, ControlTemplateStore>();
+            services.AddSingleton<IControlTemplateStore, ControlTemplateStore>(WithMinimalTemplates);
     }
 
     /// <summary>
-    /// Loads default control templates
+    /// Adds default templates to the store.
     /// </summary>
-    /// <param name="serviceProvider"></param>
-    /// <returns></returns>
-    public static ControlTemplateStore LoadControlTemplateStore(IServiceProvider serviceProvider)
+    public static ControlTemplateStore WithDefaultTemplates(IServiceProvider _)
```

Comment:
As this method is not implementing an interface, there doesn't appear to be any reason to maintain the signature by having a `IServiceProvider _` here.   Unless the intention is to eventually use the argument in a future iteration?

Ditto below for `WithMinimalTemplates(IServiceProvider _)`

Reply:
AddSingleton signature requires IServiceProvider